R/pmcmc_parameters_nested.R

```diff
@@ -247,73 +237,58 @@ pmcmc_parameters_nested <- R6::R6Class(
-    ##' corresponding to populations and columns to parameters.
-    ##'
-    ##' @param theta a parameter matrix with columns in the same order as
-    ##' `$names()` and rows in the same order as `$populations()`.
-    ##'
-    ##' @param scale an optional scaling factor to apply to the
-    ##' proposal distribution. This may be useful in sampling starting
-    ##' points. The parameter is equivalent to a multiplicative factor
-    ##' applied to the variance covariance matrix.
-    ##'
-    ##' @param type specifies which type of parameters should be proposed,
-    ##' either fixed parameters only ("fixed"), varied only ("varied"), or
-    ##' both ("both") types. For 'fixed' and 'varied', parameters of the
-    ##' other type are left unchanged.
-    propose = function(theta, scale = 1, type = c("fixed", "varied", "both")) {
+##' @description This proposes a new parameter matrix given your current
+##' matrix and the variance-covariance matrices of the proposal
+##' kernels, discretises any discrete values, and reflects bounded
+##' parameters until they lie within `min`:`max`. Returns matrix with rows
+##' corresponding to populations and columns to parameters.
+##'
+##' @param theta a parameter matrix with columns in the same order as
+##' `$names()` and rows in the same order as `$populations()`.
+##'
+##' @param scale an optional scaling factor to apply to the
+##' proposal distribution. This may be useful in sampling starting
+##' points. The parameter is equivalent to a multiplicative factor
+##' applied to the variance covariance matrix.
+##'
+##' @param type specifies which type of parameters should be proposed,
+##' either fixed parameters only ("fixed"), varied only ("varied"), or
+##' both ("both") types. For 'fixed' and 'varied', parameters of the
+##' other type are left unchanged.
+    propose = function(theta, scale = 1, type = c("both", "varied", "fixed")) {
       type <- match.arg(type)
-      pfix <- function(theta, scale) {
-        if (length(private$proposal_fixed) > 0) {
-          private$proposal_fixed(theta, scale)
-        }
+      fix_params <- private$param_names$fixed
+      var_params <- private$param_names$varied
+
+      nok <- (type == "fixed" && length(fix_params) == 0) ||
+              (type == "varied" && length(var_params) == 0)
+      if (nok) {
+        stop("Requested proposal that doesn't exist.")
       }
-      pvar <- function(theta, scale) {
-        if (length(private$proposal_varied) > 0) {
-          var <- vapply(seq(nrow(theta)), function(i)
-            private$proposal_varied[[rownames(theta)[[i]]]](theta[i, ], scale),
-            numeric(length(self$names())))
-          # catch one varied edge case
-          if (!inherits(var, "matrix")) {
-            var <- as.matrix(var)
-            rownames(var) <- private$param_names$varied
-          }
 
-          colnames(var) <- self$populations()
-          t(var)
-        }
+      # arbitrarily select first population as both same
```

Comment:
While we get this up and running, we might pop in a couple of assertions that this is true, to guard from bookkeeping failure. Can do this later, (or not at all)

Reply:
Sure I'll add these